src/Operations/ConvertContoursToMeshes.cc

```diff
@@ -527,12 +531,55 @@ bool ConvertContoursToMeshes(Drover &DICOM_data,
                 //YLOGINFO("Processing contour map from " << N_upper << " to " << N_lower);
 
                 if( (N_upper != 0) && (N_lower == 0) ){
+                    //If the upper plane contains 2 contours and one is enclosed in the other,
+                    //tile the contours together instead of closing the floor
+                    //this routine is for pipe like structures.
+                    if (N_upper == 2){
+                        if (projected_contours_overlap(*m_cp_it, pcs.upper.front(),*m_cp_it, pcs.upper.back())
+                            && !projected_contours_intersect(*m_cp_it, pcs.upper.front(),*m_cp_it, pcs.upper.back())){
+
+                            auto new_faces = Estimate_Contour_Correspondence(pcs.upper.front(), pcs.upper.back());
+                            const auto old_face_count = amesh.vertices.size();
+                            for(const auto &p : pcs.upper.front().get().points) amesh.vertices.emplace_back(p);
```

Comment:
putting vertices and faces in appears at least twice in this code, maybe create a lambda function for this